### ISSUE TITLE
Merge `src/common` into `src/frontend`

### ISSUE THREAD

**Issue by lethal-guitar** [lethal-guitar/RigelEngine#818]
Move all the files in `src/common` into `src/frontend`, adapt includes in client code, adjust `CMakeLists.txt`, update the README in `src`.

**Comment by naftalimurgor** [comment]
Hi @lethal-guitar 
Let's do this again, seems like something I can work on 🚀 

Best,
Naftali

**Comment by lethal-guitar** [comment]
Hey @naftalimurgor, welcome back! Sure thing, sounds good!

**Comment by naftalimurgor** [comment]
Greetings! @lethal-guitar 

I managed to work on this issue  and submitted a PR here https://github.com/lethal-guitar/RigelEngine/pull/831
- All checks pass except the `clang-format-check`. I totally slipped my mind that I should run `./tools/format-staged.sh` before committing. I appreciate any help resolving this 😃 
- 
- Will need to check on update to the `README.md`, I dropped the descriptions to the `common` module
